src/XR/webXRRenderTargetTextureProvider.ts

```diff
@@ -63,19 +63,22 @@ export abstract class WebXRLayerRenderTargetTextureProvider implements IWebXRRen
         // Create render target texture from the internal texture
         const renderTargetTexture = new RenderTargetTexture("XR renderTargetTexture", { width, height }, this._scene);
         const renderTargetWrapper = renderTargetTexture.renderTarget!;
-        (renderTargetWrapper as WebGLRenderTargetWrapper)._framebuffer = framebuffer;
+        // Set the framebuffer, make sure it works in all scenarios - emulator, no layers and layers
+        if (framebuffer || !colorTexture) {

```

Comment:
nit: Just curious, why is this necessary?

Reply:
This adds support to non-layers support.

I made different adjustments to the clear state of certain textures, I will see if it is still needed.